src/axmysql/projection/offsetstore.go

```diff
@@ -73,8 +73,9 @@ func (OffsetStore) SaveOffset(
 		return err
 	}
 
-	res, err := tx.ExecContext(
+	return sqlutil.UpdateSingleRow(
```

Comment:
As discussed, this whole function needs to be changed to use the `SELECT ... FOR UPDATE`, `UPDATE` pattern, because the affected row check was previously being used to perform optimistic concurrency control.

For an example, see https://github.com/jmalloc/ax/blob/fd9770ecf2d3b5cf234c75be0a707f247ecfc593/src/axmysql/saga/crud.go#L175 (from this PR). The way that the revisions are compared there is similar to how the offsets need to be compared here.